### PR TITLE
expression: implement vectorized evaluation for 'builtinNEIntSig'

### PR DESCRIPTION
**Pull request by tsthght**
### What problem does this PR solve? 
Implement vectorized evaluation for builtinNEIntSig, for #12103

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinCompareFunc/builtinNEIntSig-VecBuiltinFunc-8         	  391096	      2875 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinNEIntSig-NonVecBuiltinFunc-8      	   28605	     42841 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinNEIntSig-VecBuiltinFunc#01-8      	  414004	      2751 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinNEIntSig-NonVecBuiltinFunc#01-8   	   29678	     41874 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinNEIntSig-VecBuiltinFunc#02-8      	  369595	      3242 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinNEIntSig-NonVecBuiltinFunc#02-8   	   28356	     41751 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinNEIntSig-VecBuiltinFunc#03-8      	  361756	      3357 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinNEIntSig-NonVecBuiltinFunc#03-8   	   24622	     47705 ns/op	       0 B/op	       0 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_compare_vec.go [hunk 1]
```diff
@@ -291,11 +291,35 @@ func (b *builtinEQIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) e
...
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf1); err != nil {
+		return err
+	}
+
+	result.ResizeInt64(n, false)
+	vecCompareInt(mysql.HasUnsignedFlag(b.args[0].GetType().Flag), mysql.HasUnsignedFlag(b.args[1].GetType().Flag), buf0, buf1, result)
+	result.MergeNulls(buf0, buf1)
```
Comment (Reminiscent) [inline expression/builtin_compare_vec.go]: The code in this function is almost the same as the builtinEQIntSig.vecEvalInt. Can we merge them?

Comment (tsthght) [reply expression/builtin_compare_vec.go]: The following functions are implemented similarly. Any goole ideas?
```
builtinLTIntSig
builtinLEIntSig
builtinGTIntSig
builtinGEIntSig
builtinEQIntSig
builtinNEIntSig
```

**Pull request merged**